packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1630,13 +1633,26 @@ export class Project {
         continue;
 
       if (cache.immutable) {
-        report.reportError(MessageName.IMMUTABLE_CACHE, `${formatUtils.pretty(this.configuration, ppath.basename(entryPath), `magenta`)} appears to be unused and would marked for deletion, but the cache is immutable`);
+        report.reportError(MessageName.IMMUTABLE_CACHE, `${formatUtils.pretty(this.configuration, ppath.basename(entryPath), `magenta`)} appears to be unused and would be marked for deletion, but the cache is immutable`);
       } else {
-        report.reportInfo(MessageName.UNUSED_CACHE_ENTRY, `${formatUtils.pretty(this.configuration, ppath.basename(entryPath), `magenta`)} appears to be unused - removing`);
+        if (preferAggregateCacheInfo)
+          entriesRemoved += 1;
+        else
+          report.reportInfo(MessageName.UNUSED_CACHE_ENTRY, `${formatUtils.pretty(this.configuration, ppath.basename(entryPath), `magenta`)} appears to be unused - removing`);
+
         await xfs.removePromise(entryPath);
       }
     }
 
+    if (preferAggregateCacheInfo && entriesRemoved !== 0) {
+      report.reportInfo(
+        MessageName.UNUSED_CACHE_ENTRY,
+        entriesRemoved > 1
+          ? `${entriesRemoved} packages appeared to be unused and were removed`
+          : `one package appeared to be unused and was removed`
```

Comment:
Would make sense to print the removed filename if there's a single one, wdyt? Something like:
```suggestion
          ? `${firstFilename} and ${n} other${n === 1 ? `` : `s`} appeared to be unused and were removed`
          : `${firstFilename} appeared to be unused and was removed`
```

Reply:
We don't do that for cache hits/misses so i'd say no